src/components/catalog-filter/catalog-filter.controller.ts

```diff
@@ -0,0 +1,167 @@
+          {
+            id: index,
+            title: vendor,
+            value: vendor,
+            selected: false
+          }
+        );
+      });
+    }
+
+    if (this.ctrl.filterOnKeyword) {
+      this.ctrl.keywordFilter.values = [this.ctrl.filterOnKeyword];
+      this.constructFiltersFromModel();
+    }
+  }
+
+  public $onChanges(onChangesObj: angular.IOnChangesObject) {
+    if (onChangesObj.filterOnKeyword  && onChangesObj.filterOnKeyword.currentValue) {
+      this.resetFilterPanelModel();
+      this.ctrl.keywordFilter.values = [this.ctrl.filterOnKeyword];
+      this.constructFiltersFromModel();
+    }
+    if (onChangesObj.clearFilters  && onChangesObj.clearFilters.currentValue) {
+      this.resetFilterPanelModel();
+      this.constructFiltersFromModel();
+    }
+  }
+
+  public onKeywordKeyPress = (keyEvent: any) => {
+    if (keyEvent.which === 13 && this.ctrl.keywordFilter.value.length > 0) {
+      // store new keywoard filter value in values array
+      this.ctrl.keywordFilter.values.push(this.ctrl.keywordFilter.value);
+      // remove the keyword value to show placeholder text
+      delete this.ctrl.keywordFilter.value;
+      this.constructFiltersFromModel();
+    }
+  };
+
+  // called when filter is changed in the filter panel
+  public filterChanged = () => {
+    this.constructFiltersFromModel();
+  };
+
+  // called when filter cleared by hitting 'x' in filter results tag, or 'Clear All Filters' link
+  private onFilterChange = (appliedFilters: any, changedFilterId: string, changedFilterValue: string) => {
+    if (angular.isDefined(changedFilterId) && angular.isDefined(changedFilterValue)) {
+      this.updateFilterPanelModel(changedFilterId, changedFilterValue);
+    } else {
+      this.resetFilterPanelModel();
+    }
+    this.constructFiltersFromModel();
+  };
+
+  private createAppliedFilter(filter: any, values: any) {
+    return {
+      id: filter.id,
+      title: filter.title,
+      filterType: filter.filterType,
+      values: values
+    };
+  };
+
+  private constructFiltersFromModel() {
+    let newAppliedFilters = [];
+    _.forEach(this.ctrl.filterPanelModel, (filter: any) => {
+      var filterValues = [];
+      if (angular.isDefined(filter.values) && filter.values.length > 0) {
+        if (filter.filterType === "checkbox") {
+          // the values of the selected checkboxes are stored in a single new appliedFilter
+          _.each(filter.values, (value: any) => {
+            if (value.selected) {
+              filterValues.push(value.value);
+            }
+          });
+          if (filterValues.length > 0) {
+            newAppliedFilters.push( this.createAppliedFilter (filter, filterValues) );
+          }
+        } else {
+          // each keyword value gets a new appliedFilter
+          _.each(filter.values, (value: string) => {
+            filterValues = [value];
+            newAppliedFilters.push(this.createAppliedFilter(filter, filterValues));
```

Comment:
Suggest just this

```js
newAppliedFilters.push(this.createAppliedFilter(filter, [ value ]));
```

 and using `let filterValues = []` inside the checkbox if above. Then the local var is only needed in that small if block.